api/src/main/java/org/openmrs/layout/name/NameSupport.java

```diff
@@ -27,16 +48,144 @@ public NameSupport() {
 	}
 	
 	public static NameSupport getInstance() {
-		if (singleton == null) {
-			throw new APIException("Not Yet Instantiated");
-		} else {
-			return singleton;
+		synchronized (NameSupport.class) {
+			singleton.init();
+			if (singleton.customTemplateConfigured) {
+				return singleton;
+			} else {
+				new NameSupport();
+			}
+			
+			if (singleton == null) {
+				throw new APIException("Not Yet Instantiated");
+			} else {
+				return singleton;
+			}
+		}
+	}
+
+	private void init() {
+		if (!initialized) {
+			Context.getAdministrationService().addGlobalPropertyListener(singleton);
+			try {
+				// Get layout NameTemplate format to override the default if any
+				String layoutTemplateXml = Context.getAdministrationService().getGlobalProperty(
+					OpenmrsConstants.GLOBAL_PROPERTY_LAYOUT_NAME_TEMPLATE);
+				
+				if (StringUtils.isNotBlank(layoutTemplateXml)) {
+					// Validating... Custom NameTemplate should be in a Valid XML format
+					DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+					DocumentBuilder builder = dbf.newDocumentBuilder();
+					Document doc = builder.parse(new ByteArrayInputStream(layoutTemplateXml.getBytes(StandardCharsets.UTF_8)));
```

Comment:
@Ruhanga, I was thinking of implementing this validation, is it viable or I discard it?!

Reply:
Validation should be ok, however for invalid cases, this should rather result into a warning log and therefore falling back to the default template.